packages/feedback-react/src/BaseFeedback.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { createContext, useCallback, useEffect, useState } from "react";
+import cn from "classnames";
+import { TextArea } from "@fremtind/jkl-text-input-react";
+import { SecondaryButton } from "@fremtind/jkl-button-react";
+import { SuccessMessage } from "@fremtind/jkl-message-box-react";
+import { FeedbackOptions, FeedbackType } from "./types";
+
+export interface BaseFeedbackProps {
+    label: string;
+    onSubmit: (data: FeedbackType) => void;
+    description?: string;
+    feedbackOptions?: FeedbackOptions[];
+    successTitle?: string;
+    successMessage?: string;
+    className?: string;
+    callToActionText?: string;
+    showTextArea?: boolean;
+    textAreaLabel?: string;
+    textAreaHelpLabel?: string;
+}
+
+export const FeedbackContext = createContext<{
+    options: FeedbackOptions[];
+    value?: FeedbackOptions;
+    setValue: (next: FeedbackOptions) => void;
+}>({ options: [], setValue: () => undefined });
+
+export const BaseFeedback: React.FC<BaseFeedbackProps> = ({
+    label,
+    onSubmit,
+    description,
+    successMessage = "Det hjelper oss i arbeidet med å forbedre våre løsninger",
+    successTitle = "Takk for tilbakemeldingen!",
+    callToActionText = "Send inn tilbakemelding",
+    showTextArea = true,
+    textAreaLabel = "Tips oss gjerne om hva vi kan forbedre",
+    textAreaHelpLabel = "",
+    className = "",
+    feedbackOptions = [1, 2, 3, 4, 5],
+    children,
+}) => {
+    const [feedbackValue, setFeedbackValue] = useState<FeedbackOptions>();
+    const [message, setMessage] = useState("");
+    const [feedbackSubmitted, setFeedbackSubmitted] = useState(false);
+
+    const handleSubmit = useCallback(() => {
+        if (!feedbackSubmitted && feedbackValue) {
+            onSubmit({ feedbackValue, message });
+        }
+    }, [feedbackValue, feedbackSubmitted, message, onSubmit]);
+
+    const handleActiveSubmit = () => {
+        setFeedbackSubmitted(true);
+        handleSubmit();
+    };
+
+    useEffect(() => {
+        if (typeof window !== "undefined") {
+            window.addEventListener("beforeunload", handleSubmit);
+        }
+        return () => {
+            window.removeEventListener("beforeunload", handleSubmit);
+        };
+    }, [handleSubmit]);
+
+    if (feedbackSubmitted) {
+        return <SuccessMessage title={successTitle}>{successMessage}</SuccessMessage>;
+    }
+
+    return (
+        <FeedbackContext.Provider
+            value={{ options: feedbackOptions, setValue: setFeedbackValue, value: feedbackValue }}
+        >
+            <form className={`jkl-feedback ${className}`} onSubmit={(e) => e.preventDefault()}>
+                <header className="jkl-feedback__heading">
+                    <h2 className="jkl-heading-large">{label}</h2>
```

Comment:
Tror ikke vi burde ha denne som et definert overskriftsnivå. Blir _veldig_ fort gjort at den ikke matcher hierarkiet på siden.. Holder det ikke at den er inne i et `<header>`-element? Eventuelt kanskje man skulle hatt en prop for å velge nivå eller elementtype?

Selve skjemaet bør nok også ha en rolle av noen sort (f.eks. [region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Region_role)) for å sørge for at `<header>`-en "hører til" riktig del av siden.

Reply:
Så ikke denne ved første gjennomgang, så beklager vinglete tilbakemelding 😅

Reply:
Enig i semantikken.

Men holder det ikke at den er i et <form> burde vel ikke trenge en seksjon rundt det?

Reply:
dropper det semantiske elementet, form burde være nok her.